File: src/main/java/com/pengrad/telegrambot/model/UserProfilePhotos.java
package com.pengrad.telegrambot.model;

import java.util.Arrays;

/**
 * stas
 * 8/5/15.
 */
public class UserProfilePhotos {

    private Integer total_count;
    private PhotoSize[][] photos;

    UserProfilePhotos() {
    }

    public Integer totalCount() {
        return total_count;
    }

    public PhotoSize[][] photos() {
        return photos;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        UserProfilePhotos that = (UserProfilePhotos) o;

        if (total_count != null ? !total_count.equals(that.total_count) : that.total_count != null) return false;
        return Arrays.deepEquals(photos, that.photos);
    }

    @Override
    public int hashCode() {
        int result = total_count.hashCode();
        result = 31 * result + Arrays.deepHashCode(photos);
        return result;
    }

    @Override
    public String toString() {
        return "UserProfilePhotos{" +
                "total_count=" + total_count +
                ", photos=" + Arrays.toString(photos) +
                '}';
    }
}

File: core/src/commonMain/kotlin/com/pengrad/telegrambot/model/UserProfilePhotos.kt
package com.pengrad.telegrambot.model

import java.io.Serializable
import java.util.Arrays

/**
 * stas
 * 8/5/15.
 */
class UserProfilePhotos : Serializable {

    private val total_count: Int? = null
    private val photos: Array<Array<PhotoSize>>? = null

    fun totalCount(): Int? {
        return total_count
    }

    fun photos(): Array<Array<PhotoSize>>? {
        return photos
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as UserProfilePhotos?

        return if (if (total_count != null) total_count != that!!.total_count else that!!.total_count != null) false else Arrays.deepEquals(photos, that.photos)
    }

    override fun hashCode(): Int {
        var result = total_count?.hashCode() ?: 0
        result = 31 * result + Arrays.deepHashCode(photos)
        return result
    }

    override fun toString(): String {
        return "UserProfilePhotos{" +
                "total_count=" + total_count +
                ", photos=" + Arrays.deepToString(photos) +
                '}'.toString()
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
